actors/runtime/src/runtime/fvm.rs

```diff
@@ -263,28 +261,40 @@ where
         value: TokenAmount,
     ) -> Result<RawBytes, ActorError> {
         if self.in_transaction {
-            return Err(actor_error!(SysErrIllegalActor; "runtime.send() is not allowed"));
+            return Err(actor_error!(unspecified; "runtime.send() is not allowed"));
         }
-        // TODO: distinguish between "syscall" errors and actor exit codes.
         match fvm::send::send(&to, method, params, value) {
             Ok(ret) => {
                 if ret.exit_code.is_success() {
                     Ok(ret.return_data)
                 } else {
-                    Err(ActorError::from(ret.exit_code))
+                    // The returned code can't be simply propagated as it may be a system exit code.
+                    Err(ActorError::internal_send_aborted(format!(
+                        "send to {} method {} aborted with code {}",
+                        to, method, ret.exit_code
+                    )))
                 }
             }
             Err(err) => Err(match err {
+                // Some of these errors are from operations in the Runtime or SDK layer
+                // before or after the underlying VM send syscall.
                 ErrorNumber::NotFound => {
-                    actor_error!(SysErrInvalidReceiver; "receiver not found")
+                    // How do we know the receiver is what triggered this?
+                    // An error number doesn't carry enough information at this level
+                    // above the raw syscall (and even then).
+                    actor_error!(not_found; "receiver not found")
                 }
                 ErrorNumber::InsufficientFunds => {
-                    actor_error!(SysErrInsufficientFunds; "not enough funds")
+                    // Actually this is more like an illegal argument where the caller attempted
+                    // to transfer an amount larger than that representable by the VM's
+                    // token amount type. Yes, the caller doesn't have that amount, but if they'd
+                    // attempted to transfer a representable amount it would fail with
+                    // SYS_INSUFFICIENT_FUNDS instead, so this difference is wierd.
```

Comment:
It'll fail with `InsufficientFunds` (the syscall error) in both cases. See https://github.com/filecoin-project/fvm-project/blob/724650bf92b239bb7d2079cde7d5bad121d7c79f/07-errors.md#L53-L57 for the only "system" exit codes that can appear here.

Basically, the rules are:

1. Syscall error numbers indicates that something went wrong with the syscall.
2. Exit codes indicates that the _syscall_ succeeded, but the called actor failed or did something wrong.

In other words:

1. If we fail _before_ we invoke the target actor, we get a syscall error.
2. If we fail afterwards, we get an exit code. Usually a "user" exit code, but sometimes a "system" exit code if the actor fails without yielding a usable exit code.

Other than that, the _system_ exit codes exist for recording failures on-chain.